FinSearchAPI/Helper/StaticHelper.php

```diff
@@ -70,33 +67,50 @@ public static function getProductsFromXml( SimpleXMLElement $xmlResponse ) {
 		try {
 			$container = Shopware()->Container();
 			/** @var StoreFrontBundle\Service\ProductNumberServiceInterface $productService */
-			$productService = $container->get('shopware_storefront.product_number_service');
+			$productService = $container->get( 'shopware_storefront.product_number_service' );
 			/* READ PRODUCT IDS */
 			foreach ( $xmlResponse->products->product as $product ) {
 				try {
+
 					$articleId = (string) $product->attributes()['id'];
-					$productCheck = $productService->getMainProductNumberById($articleId);
-					if ($articleId === '' || $articleId === null){
+
+					$productCheck = $productService->getMainProductNumberById( $articleId );
+
+					if ( $articleId === '' || $articleId === null ) {
 						continue;
 					}
 					/** @var array $baseArticle */
-					$baseArticle = Shopware()->Modules()->Articles()->sGetArticleById( $articleId );
-					if ( $baseArticle !== null && count( $baseArticle ) > 0 ) {
-						$foundProducts[] = $baseArticle;
-					}
+					$baseArticle                = array();
+					$baseArticle['orderNumber'] = $productCheck;
+					$baseArticle['detailId']    = self::getDetailIdForOrdernumber( $productCheck );
+					$foundProducts[ $articleId ] = $baseArticle;
 				} catch ( Exception $ex ) {
+				//	die($ex->getMessage());
 					// No Mapping for Search Results
+					continue;
 				}
 			}
 		} catch ( Exception $ex ) {
 			// Logging Function
 			//print_r($ex->getMessage());
-
 		}
 
 		return $foundProducts;
 	}
 
+	public static function getDetailIdForOrdernumber( string $ordernumber ) {

```

Comment:
Ich meine mich zu erinnern, dass es in ProductNumberService schon so eine Methode gibt.